### PR TITLE
Add demesh

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/cgal.cc [hunk 1]
```diff
@@ -1205,6 +1272,112 @@ const Surface_mesh* ApproximateSurfaceMesh(
...
+       e = m.next_around_source(e)) {
+    if (!is_coplanar_edge(m, p, e) && !is_collinear_edge(m, p, start, e)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+template <typename Kernel>
+class Demesh_cost {
```
Comment (BarbourSmith) [inline algorithm/cgal/cgal.cc:1308]: How did the cost come out? Is it expensive?

File: geometry/graph/remesh.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { toConcreteGeometry } from './../tagged/toConcreteGeometry.js';
 import { toSurfaceMesh } from './toSurfaceMesh.js';
 
 export const remesh = (geometry, options = {}, selections = []) => {
-  const { method = 'edgeLength', lengths = [1] } = options;
+  const { method = 'isotropic', lengths = [1] } = options;
```
Comment (BarbourSmith) [inline geometry/graph/remesh.js:13]: 👍

**Review by BarbourSmith — APPROVED** [review]
This one is quite a bit bigger than I expected. Looks good to me!

File: algorithm/cgal/cgal.cc (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/cgal.cc:1308]: The cost is used to prioritize the edge collapses.

**Pull request merged**